### PR TITLE
refactor(styleoverride): remove styleOverride props from carbon (FE-3776)

### PR DESCRIPTION
**Pull request by nicktitchmarsh**
### Proposed behaviour
Remove deprecated styleOverride props

### Checklist

- [x] Commits follow our style guide
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [ ] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

**Review by sjohnson-sage — COMMENTED** [review]
Looks good, just a few things I think you've missed: 

- ActionPopover styleOverride Cypress tests -> `/cypress/features/regression/test/actionPopoverStyleOverrideNoIFrame.feature`
- ActionPopover `styles overriden story`
- AnchorNavigation `with overriden styles` story

File: src/__experimental__/components/input/input-presentation.component.js [hunk 1]
```diff
@@ -82,7 +65,6 @@ InputPresentation.propTypes = {
   error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
   warning: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
   info: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
-  styleOverride: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
```
[the remark this answers is not in the document]
Comment (nicktitchmarsh) [reply src/__experimental__/components/input/input-presentation.component.js:85]: These should be removed now. Thanks

**Review by ianforster-sage — COMMENTED** [review]
Please could you double-check that these lines are ok to be left in the code:

https://github.com/Sage/carbon/blob/9c36b58d9ffc13686bc526660507f78ffeebd7da/src/components/anchor-navigation/anchor-navigation.stories.js#L142

https://github.com/Sage/carbon/blob/9c36b58d9ffc13686bc526660507f78ffeebd7da/src/__experimental__/components/textbox/textbox.stories.js#L25

**Review by ianforster-sage — APPROVED** [review]

**Review by sjohnson-sage — COMMENTED** [review]
Looks good to me, other than the commits need to be breaking changes - I'll approve once you've done that 👍🏻

**Review by sjohnson-sage — DISMISSED** [review]

**Review by ianforster-sage — DISMISSED** [review]

File: src/__experimental__/components/fieldset/fieldset-examples.stories.mdx [hunk 2]
```diff
@@ -55,7 +55,7 @@ Address Fieldset for Canada.
         label='ZIP Code'
         labelInline
         labelAlign='right'
-        styleOverride={ { input: { width: '120px', flex: 'none' } } }
+        inputWidth={10}
```
Comment (ianforster-sage) [inline src/__experimental__/components/fieldset/fieldset-examples.stories.mdx:58]: 10*8px = 80px
What's the story here?

File: src/__experimental__/components/fieldset/fieldset-examples.stories.mdx [hunk 3]
```diff
@@ -80,7 +80,7 @@ Address Fieldset for EU.
         label='Postal Code'
         labelInline
         labelAlign='right'
-        styleOverride={ { input: { width: '120px', flex: 'none' } } }
+        inputWidth={10}
```
Comment (ianforster-sage) [inline src/__experimental__/components/fieldset/fieldset-examples.stories.mdx:83]: ditto

File: src/components/accordion/accordion.d.ts [hunk 4]
```diff
@@ -28,6 +28,10 @@ export interface AccordionProps extends SpaceProps {
   iconType?: "chevron_down" | "dropdown";
   /** Sets icon alignment - accepted values: 'left', 'right' (default) */
   iconAlign?: AlignBinaryType;
+  /** Sets accordion type to either primary (default), or secondary */
+  type?: ThemesBinary;
```
Comment (sjohnson-sage) [inline src/components/accordion/accordion.d.ts:32]: Did you mean to add these or is it from the rebase?

File: src/__experimental__/components/fieldset/fieldset-examples.stories.mdx (hunk 2 above)
Comment (nicktitchmarsh) [reply src/__experimental__/components/fieldset/fieldset-examples.stories.mdx:58]: The old input width included the label. This is the actual input width.

File: src/__experimental__/components/fieldset/fieldset-examples.stories.mdx (hunk 3 above)
Comment (nicktitchmarsh) [reply src/__experimental__/components/fieldset/fieldset-examples.stories.mdx:83]: The old input width included the label. This is the actual input width.

File: src/components/accordion/accordion.d.ts (hunk 4 above)
Comment (nicktitchmarsh) [reply src/components/accordion/accordion.d.ts:32]: Thats from a rebase, I did not add that. I can reverse it out though

**Review by ianforster-sage — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

File: src/components/accordion/accordion.d.ts [hunk 5]
```diff
@@ -42,18 +44,9 @@ export interface AccordionProps extends SpaceProps {
...
-    headerArea?: () => object | object;
-    icon?: () => object | object;
-    header?: () => object | object;
-    content?: () => object | object;
-  };
   /** Sets accordion sub title */
   subTitle?: string;
   /** Sets accordion title */
-  title: string;
-  /** A warning message to be displayed in the tooltip */
```
Comment (sjohnson-sage) [inline src/components/accordion/accordion.d.ts:56]: You've deleted the wrong comment here

**Review by sjohnson-sage — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**